internal/controller/todo.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/daniyalumer/todo-list-go-chi/internal/helper"
+	"github.com/daniyalumer/todo-list-go-chi/internal/models"
+	"github.com/daniyalumer/todo-list-go-chi/internal/service"
+	"github.com/go-chi/chi/v5"
+)
+
+func TodoRoutes() chi.Router {
+	r := chi.NewRouter()
+
+	r.Get("/", GetTodosHandler)
+	r.Post("/", CreateTodoHandler)
+	r.Put("/", UpdateTodoHandler)
+	r.Delete("/", DeleteTodoHandler)
+
+	return r
+}
+
+func GetTodosHandler(w http.ResponseWriter, r *http.Request) {
+	todos, err := service.ReadTodoList()
+	if err != nil {
+		helper.TodoListResponseWriter(w, "failed to read todo list", []models.Todo{}, http.StatusBadRequest)
+		return
+	}
+	helper.TodoListResponseWriter(w, "todo list read successfully", todos, http.StatusAccepted)
+}
+
+func CreateTodoHandler(w http.ResponseWriter, r *http.Request) {
+	if !helper.ParseForm(w, r) {
+		return
+	}
+
+	todoDescription := r.FormValue("description")
+	userIDStr := r.URL.Query().Get("userId")
+
+
+	userID, err := strconv.Atoi(userIDStr)
+	if err != nil {
+		http.Error(w, "Invalid ID", http.StatusBadRequest)
+		return
+	}
+
+	todo, err := service.CreateTodo(todoDescription, userID)
+	if err != nil {
+		helper.TodoResponseWriter(w, "failed to create todo item", models.Todo{}, http.StatusBadRequest)
```

Comment:
Please explain the reason behind mentioning the bad request status code?

Reply:
It should not be a bad request since there is no request body. It should instead be `http.InternalServerError`